api/envoy/service/discovery/v3/discovery.proto

```diff
@@ -241,4 +253,14 @@ message Resource {
 
   // The resource being tracked.
   google.protobuf.Any resource = 2;
+
+  // [#not-implemented-hide:]
+  // Time-to-live value for the resource. For each resource, a timer is started. The timer is reset each time the
```

Comment:
A per-resource timer seems like a very heavy-weight solution.  Is there no way to do this less granularly?

Reply:
For incremental xDS, this seems like the most intuitive way to do it. It would be possible to remove all resources on TTL expiration, as will be done with SotW, but that seems less intuitive in the delta implementation, IMO.

Reply:
I think this is also an opt-in feature. If you are super performance sensitive you can opt not to use TTL.